detection_rules/kbwrap.py

```diff
@@ -107,3 +111,54 @@ def search_alerts(ctx, query, date_range, columns, extend):
         table_columns = table_columns + columns if extend else columns
     click.echo(Table.from_list(table_columns, alerts))
     return alerts
+
+
+@kibana_group.group('experimental')
+def kibana_experimental():
+    """[Experimental] helper commands for integrating with Kibana."""
+    click.secho('\n* experimental commands are use at your own risk and may change without warning *\n')
+
+
+@kibana_experimental.command('create-dnstwist-rule')
+@click.argument('input-file', type=click.Path(exists=True, dir_okay=False), required=True)
+@click.pass_context
+def create_dnstwist_rule(ctx: click.Context, input_file, verbose=True):
+    """Create an indicator match or query rule based on dnstwist results."""
+    kibana = ctx.obj['kibana']
+
+    root_dir = Path(__file__).parent.parent
+    rule_template_file = root_dir / 'etc' / 'rule_template_typosquatting_domain.toml'
+    custom_rule_file = (
+            root_dir / "rules" / "cross-platform" / "initial_access_dns_request_for_typosquatting_domain.toml"
+    )
+
+    click.echo(f'Attempting to load dnstwist data from {input_file}')
+    dnstwist_data = load_dump(input_file)
+
+    watchlist_domains = []
+
+    for record in dnstwist_data:
+        if record['fuzzer'] == 'original*':
+            original_domain = record['domain-name']
+            click.echo(f'Original domain name identified: {original_domain}')
+        else:
+            watchlist_domains.append(record['domain-name'])
+
+    click.echo(f'{len(watchlist_domains)} watchlist domains identified')
+
+    click.echo('Attempting to create threat match rule')
+    template_rule = load_dump(str(rule_template_file))
+
+    # Populate template rule with user's custom info
+    template_rule['metadata']['creation_date'] = datetime.date.today().strftime("%Y/%m/%d")
+    template_rule['metadata']['updated_date'] = datetime.date.today().strftime("%Y/%m/%d")
+    template_rule['rule']['author'].append(click.prompt('Enter rule author'))
+    template_rule['rule_id'] = str(uuid4())
+
+    # Create rule object and validate it against schema
+
+    # Save rule in toml format
+    # click.echo(f'Saving rule to {custom_rule_file}')
+    # save_dump(template_rule, str(custom_rule_file))
```

Comment:
@brokensound77 do you know if it's possible for me to create a threat match rule object from a dict and validate it against the schema before saving it to a toml file? 🤔

Reply:
don't go dict but straight to dataclass if you can